Control/Monad/Logic.hs

```diff
@@ -83,12 +83,31 @@ observeT :: MonadFail m => LogicT m a -> m a
 observeT lt = unLogicT lt (const . return) (fail "No answer.")
 
 -------------------------------------------------------------------------
--- | Extracts all results from a LogicT computation.
+-- | Extracts all results from a @LogicT@ computation, unless blocked by the
+-- underlying monad.
+--
+-- For example, given
+-- 
+-- >>> let nats = pure 0 `mplus` ((1 +) <$> nats)
+--
+-- some monads (like @Identity@, @Reader@, @Writer@, and @State@)
+-- will be productive
+-- 
+-- >>> take 5 $ runIdentity $ observeAllT nats
+-- [0,1,2,3,4]
+-- 
+-- but others (like @ExceptT@, and @ContT@) will not
+-- 
+-- @fmap (take 20) $ runExcept $ observeAllT nats@
```

Comment:
```suggestion
-- >>> take 20 <$> runExcept (observeAllT nats)
```

Reply:
I'm a little concerned with putting nonterminating code after `>>>` since some tools will try to run it.  But we can always change it back if it's an issue.

Reply:
I think it might be making the travis build hang?